src/core/public/http/http_setup.ts

```diff
@@ -100,48 +100,51 @@ export const setup = (
 
   // Request/response interceptors are called in opposite orders.
   // Request hooks start from the newest interceptor and end with the oldest.
-  function interceptRequest(
+  async function interceptRequest(
     request: Request,
     controller: HttpInterceptController
   ): Promise<Request> {
     let next = request;
 
-    return [...interceptors].reduceRight(
-      (promise, interceptor) =>
-        promise.then(
-          async (current: Request) => {
-            checkHalt(controller);
-
-            if (!interceptor.request) {
-              return current;
-            }
+    try {
+      return await [...interceptors].reduceRight(
+        (promise, interceptor) =>
+          promise.then(
+            async (current: Request) => {
+              next = current;
```

Comment:
Why are we setting `next = current` here? If I remove this line, the https://github.com/elastic/kibana/blob/6b76917802148482f12fa83cdeb91533604bab95/src/core/public/http/http_service.test.ts tests still pass.

Reply:
It should still be tracking the latest value in case an error occurs. This behavior could probably be expanded upon in the tests, where it would probably fail upon removal.

Reply:
If this solves some expected behavior, there should be a test that ensures we don't break it.